unit-tests/heitech.zer0mqXt.core.tests/RequestReplyTests.cs

```diff
@@ -138,6 +138,25 @@ public async Task AsyncRqRep()
             Assert.True(result.IsSuccess);
         }
 
+        [Fact]
+        public async Task Exception_propagation_when_server_response_Throws_to_Requester()
+        {
+             var ipc = new ConfigurationTestData().GetSocketConfigInProc;
+            var sut = new Socket(ipc);
+            sut.Respond<Request, Response>(r =>
+            {
+                throw new ArgumentException("this is a unit test proving the exception propagation works");
+            });
+
+            // Act
+            var result = await sut.RequestAsync<Request, Response>(new Request { RequestNumber = 2 });
+
+            // Assert
+            Assert.False(result.IsSuccess);
+            Assert.Contains("ArgumentException", result.Exception?.Message ?? "");
```

Comment:
I'm a bit unsure about the use of _null-conditional_ and _null-coalescing_ operators here:
Could the `Exception` be null here?
Is it because if the test fails, it doesn't fail with an `NullReferenceException`, but with an assertion-exception from the testing-framework?

Reply:
This is a remainder of some testing. Will fix it.

Reply:
Awesome, I like the explicit assertion.